catalog/spatial/stanag4559/spatial-stanag4559-source/src/test/java/org/codice/ddf/spatial/ogc/stanag4559/catalog/source/client/Client.java

```diff
@@ -0,0 +1,429 @@
+                new String[0],
+                new SortAttribute[0],
+                new NameValue[0]);
+        submitQueryRequest.complete_DAG_results(dagListHolder);
+        System.out.println("Server Responded with " + dagListHolder.value.length + " result(s).\n");
+        return dagListHolder.value;
+    }
+
+    private static void processAndPrintResults(DAG[] results) {
+        System.out.println("Printing DAG Attribute Results...");
+        for (int i = 0; i < results.length; i++) {
+            printDAGAttributes(results[i]);
+            retrieveProductFromDAG(results[i]);
+        }
+    }
+
+    private static void printDAGAttributes(DAG dag) {
+        for (int i = 0; i < dag.nodes.length; i++) {
+            Node node = dag.nodes[i];
+            if (node.node_type.equals(NodeType.ATTRIBUTE_NODE)) {
+
+                String result = null;
+
+                if (node.value.type()
+                        .toString()
+                        .contains("unbounded string")) {
+                    result = node.value.extract_string();
+                } else if (node.value.type()
+                        .toString()
+                        .contains("ulong")) {
+                    result = "" + node.value.extract_ulong();
+                } else if (node.value.type()
+                        .toString()
+                        .contains("double")) {
+                    result = "" + node.value.extract_double();
+                } else if (node.value.type()
+                        .toString()
+                        .contains("AbsTime")) {
+                    AbsTime absTime = AbsTimeHelper.extract(node.value);
+                    String absDate =
+                            absTime.aDate.month + "/" + absTime.aDate.day + "/" + absTime.aDate.year
+                                    + ":";
+                    String absHour = absTime.aTime.hour + ":" + absTime.aTime.minute + ":"
+                            + absTime.aTime.second;
+                    result = absDate + absHour;
+                }
+                System.out.printf("%25s : %s\n", node.attribute_name, result);
+            }
+        }
+        System.out.println();
+    }
+
+    private static void retrieveProductFromDAG(DAG dag) {
+        System.out.println("Downloading products...");
+        for (int i = 0; i < dag.nodes.length; i++) {
+            Node node = dag.nodes[i];
+            if (node.attribute_name.equals("productUrl")) {
+
+                String url = node.value.extract_string();
+                String productPath = "product.jpg";
+                System.out.println("Downloading product : " + url);
+                try {
+                    URL fileDownload = new URL(url);
+                    BufferedInputStream inputStream =
+                            new BufferedInputStream(fileDownload.openStream());
+                    FileOutputStream outputStream = new FileOutputStream(new File(productPath));
+
```

Comment:
Definitely use try with resources where you can and then finally blocks if something doesn't implement auto closeable.